### ISSUE TITLE
cabal configure doesn't generate cabal_macros.h

### ISSUE THREAD

**Issue by mpickering** [haskell/cabal#2825]
If I remember correctly, `cabal configure` used to generate the macros file and place it in `dist/build/autogen/` with a recent build of cabal this doesn't seem to happen. Is this a recent change or am I misremembering how `cabal configure` used to behave? This file is very useful for tool writers and quite difficult to generate via the API.

**Comment by DanielG** [comment]
I don't think it ever did that, we used to have a very dirty workaround for that in ghc-mod that calls `cabal build` and then kills it once that file appears. It's not actually difficult to generate, a call to `initialBuildSteps` will do it.

**Comment by 23Skidoo** [comment]
Changing the status of this ticket to `enhancement`, should be possible to do this at configuration time.
